packages/sdk/src/manifold/types.ts

```diff
@@ -0,0 +1,81 @@
+export enum ListingType {
+  INVALID,
+  INDIVIDUAL_AUCTION,
+  FIXED_PRICE,
+  DYNAMIC_PRICE,
+  RANKED_AUCTION,
+}
+
+export enum Spec {
+  NONE,
+  ERC721,
+  ERC1155,
+}
+
+export type BNHex = {
+  type: string;
+  hex: string;
+};
+
+type BaseListing = {
+  id: string;
+  seller: string;
+  fees: DeliveryFees;
+};
+
+export type DeliveryFees = {
+  deliverFixed: number | null;
+  deliverBPS?: number;
```

Comment:
Let's have all of these mandatory, and instead of `null` and `undefined` use `0` as the default (which is what the exchange contract uses).